jsf/subsystem/src/test/java/org/jboss/as/jsf/deployment/JSFModuleIdFactoryTestCase.java

```diff
@@ -52,25 +54,44 @@ public void validSlotTest() {
     @Test
     @Ignore("Depends on https://issues.redhat.com/browse/WFLY-17405")
     public void modIdsTest() {
-        Assert.assertEquals(API_MODULE, factory.getApiModId("main").getName());
-        Assert.assertEquals("main", factory.getApiModId("main").getSlot());
-        Assert.assertEquals(IMPL_MODULE, factory.getImplModId("main").getName());
-        Assert.assertEquals("main", factory.getImplModId("main").getSlot());
-        Assert.assertEquals(INJECTION_MODULE, factory.getInjectionModId("main").getName());
-        Assert.assertEquals("main", factory.getInjectionModId("main").getSlot());
-
-        Assert.assertEquals(API_MODULE, factory.getApiModId("myfaces").getName());
-        Assert.assertEquals("myfaces", factory.getApiModId("myfaces").getSlot());
-        Assert.assertEquals(IMPL_MODULE, factory.getImplModId("myfaces").getName());
-        Assert.assertEquals("myfaces", factory.getImplModId("myfaces").getSlot());
-        Assert.assertEquals(INJECTION_MODULE, factory.getInjectionModId("myfaces").getName());
-        Assert.assertEquals("myfaces", factory.getInjectionModId("myfaces").getSlot());
-
-        Assert.assertEquals(API_MODULE, factory.getApiModId("myfaces2").getName());
-        Assert.assertEquals("myfaces2", factory.getApiModId("myfaces2").getSlot());
-        Assert.assertEquals(IMPL_MODULE, factory.getImplModId("myfaces2").getName());
-        Assert.assertEquals("myfaces2", factory.getImplModId("myfaces2").getSlot());
-        Assert.assertEquals(INJECTION_MODULE, factory.getInjectionModId("myfaces2").getName());
-        Assert.assertEquals("myfaces2", factory.getInjectionModId("myfaces2").getSlot());
+        // Canonicalize module names using ModuleIdentifierUtil
+        String mainCanonical = canonicalModuleIdentifier("main", "main");
+        String myfacesCanonical = canonicalModuleIdentifier("myfaces", "myfaces");
+        String myfaces2Canonical = canonicalModuleIdentifier("myfaces2", "myfaces2");
+
+        // Test API module for "main"
+        String apiModIdMain = factory.getApiModId(mainCanonical);
+        String implModIdMain = factory.getImplModId(mainCanonical);
+        String injectionModIdMain = factory.getInjectionModId(mainCanonical);
+
+        Assert.assertEquals(API_MODULE, apiModIdMain); // Direct string comparison
+        Assert.assertEquals("main", mainCanonical);  // The slot for "main" should be "main"
+        Assert.assertEquals(IMPL_MODULE, implModIdMain);
+        Assert.assertEquals("main", mainCanonical);
+        Assert.assertEquals(INJECTION_MODULE, injectionModIdMain);
+        Assert.assertEquals("main", mainCanonical);
+
+        // Repeat for "myfaces" and "myfaces2"
+        String apiModIdMyfaces = factory.getApiModId(myfacesCanonical);
+        String implModIdMyfaces = factory.getImplModId(myfacesCanonical);
+        String injectionModIdMyfaces = factory.getInjectionModId(myfacesCanonical);
+
+        Assert.assertEquals(API_MODULE, apiModIdMyfaces);
```

Comment:
This should be:

Assert.assertEquals(canonicalModuleIdentifier(API_MODULE, "myfaces"), apiModIdMyfaces);

The same pattern applies to all the similar asserts below.